Feature - remove all resource group locks (#96)
arcus-azure/arcus.scripting: docs/preview/features/powershell/arm.md

This module provides the following capabilities:
- [Injecting content into an ARM template](#injecting-content-into-an-arm-template)

> - [Removing resource locks on azure](#removing-resource-locks-on-azure)

Reviewer: ```suggestion
- [Removing resource locks within an Azure Resource Group](#removing-resource-locks-within-an-azure-resource-group)
```

> - [Removing resource locks within an Azure Resource Group](#removing-resource-locks-within-an-azure-resource-group)

Reviewer: This line can be removed from this document.

Author: Woops, thanks!

**Recommendations**
Always inject the content in your ARM template as soon as possible, preferably during release build that creates the artifact

> ## Removing resource locks on Azure

Reviewer: ```suggestion
## Removing resource locks within an Azure Resource Group
```

In some deployments resource-locks are being assigned. To help in removing these quickly, we have provided you with a function that removes all the existing locks on the resource group.

> While this seems dangerous, only users with sufficient access rights are allowed to delete locks.

Reviewer: Perhaps include some information regarding the specific access rights required:
```suggestion
While this seems dangerous, only users with access to the following actions are allowed to delete locks:
 - _Microsoft.Authorization/*_
 - _Microsoft.Authorization/locks/*_
 
 Of the built-in roles, only **Owner** and **Access Administrator** are granted those actions.
```

Author: Ok, thank you for this!

| Parameter           | Mandatory | Description                                                                                       |
| ------------------- | --------- | ------------------------------------------------------------------------------------------------- |
| `ResourceGroupName` | yes       | The name of the resource group where the locks should be removed                                  |
| `LockName`          | no        | The optional name of the lock to remove. When this is not provided, all the locks will be removed |

**Usage*

> When you want to remove all the resource locks, no matter what the name or the sub-location:

Reviewer: ```suggestion
When you want to remove all the resource locks within a specific resource group, no matter what the name or the sub-location:
```

```powershell
PS> Remove-AzResourceGroupLocks -ResourceGroupName "your-resource-group-name"
# Retrieving all locks in resourceGroup 'your-resource-group-name'
# Start removing all locks in resourceGroup 'your-resource-group-name'
# All locks in resourceGroup 'your-resource-group-name' have been removed
```

> When you want to remove a specific resource lock, with a given name:

Reviewer: ```suggestion
When you want to remove a specific resource lock within a given resource group, with a given name:
```
